docs/KUBERNETES.md

```diff
@@ -0,0 +1,201 @@
+
+## Uninstalling
+
+The default configuration places all of the Kubernetes Objects in a unique namespace. To completely remove all traces of the agent you need to simply delete this namespace:
+
+```console
+kubectl delete -f k8s/agent-namespace.yaml
+```
+
+If you're sharing the namespace with other applications, you can also remove all traces of the agent by deleting with a label filter. You'll also need to remove the logdna-agent-key secret which doesn't have a label:
+
+```console
+kubectl api-resources --verbs=list --namespaced -o name | xargs -n 1 kubectl get --show-kind --ignore-not-found -n <NAMESPACE> -l app.kubernetes.io/name=logdna-agent -o name | xargs kubectl delete -n <NAMESPACE>
+kubectl delete secret -n <NAMESPACE> logdna-agent-key
+```
+
+## Run as Non-Root
+
+By default the agent is configured to run as root. If this behavior is not desired, the DaemonSet can be modified to run the agent as a non-root user.
+
+This is accomplished through Linux Capabilities that makes the agent a "Capability-dumb binary." Specifically, the agent is only allowed global read access with `CAP_DAC_READ_SEARCH`. This capability is already baked into the image and configuration. The only required step is configuring the agent DaemonSet to run as the user and group `5000:5000`.
+
+To update the local configuration file, add two new fields, `runAsUser` and `runAsGroup`, to the `securityContext` section found in the `logdna-agent` container in the `logdna-agent` DaemonSet inside of `k8s/agent-resources.yaml` [`spec.template.spec.containers.0.securityContext`]:
+
+```yaml
+securityContext:
+  runAsUser: 5000
+  runAsGroup: 5000
+```
+
+Apply the new updated configuration to your cluster:
+
+```console
+kubectl apply -f k8s/agent-resources.yaml
+```
+
+Alternatively, to update the DaemonSet configuration directly in your cluster use the following patch command:
+
+```console
+kubectl patch daemonset -n logdna-agent logdna-agent --type json -p '[{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsUser","value":5000},{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsGroup","value":5000}]'
+```
+
+## Collecting Node Journald Logs
+
+The agent by default only captures logs generated by the containers running on the Kubernetes clusters container runtime environment. It does not; however, collect system component logs from applications running directly on the node such as the kubelet and container runtime. With some configuration on both the node and the agent, these journald logs can be exposed from the node to the agent.
+
+The agent can access Journald logs from the host node by mounting the logs from `/var/log/journald`. This requires enabling journald log storage in the node as well as configuring the agent to monitor the directory.
+
+### Enabling Journald on the Node
+
+Follow the steps below to ensure journald logs are written to `/var/log/journald`:
+1. Gain root access to your node
```

Comment:
by root access you mean at least global read right?

Reply:
This doesn't pertain to the agent's container and running as a non-root user. This is for administrators to directly enable journald logging on the node itsself. They will require READ+WRITE permissions for some systemd configuration files and root is the easiest option to get that.

Reply:
Simple enough